### DISCUSSION TITLE
Is it possible to get the coordinate where a user clicks in the map?

### DISCUSSION THREAD

**Discussion by ATL2001** [developmentseed/lonboard#633] [Q&A] [answered]
I'd like to be able to get the coordinate of a user's click in the map, no matter if there is a feature on the map there or not. I've attached a simple function the map's .observe() which prints the change, and it fires when I pan/zoom, but I'm not seeing anything come across when I click on the map.  

Is there a way to get a coordinate back from a click in the map?  if not is there a plan to add such functionality? I imagine if it doesn't currently exist that it would require some javascript to get, which isn't my native language, but I'd love to try to help out to make it possible.  

I've been truly impressed with my lonboarding thus far, keep up the great work everyone!

**Comment by kylebarron** [comment, marked as answer]
This is not currently possible. We don't expose an `on_click` handler, though we could. 

The JS side exposes the screen coordinate clicked, and then we'd have to map back to geospatial coordinate for the user.

If you're interested in contributing this, I can point you to how it could be implemented.

**Reply by ATL2001** [reply]
I don't have a lot of free time right now, but if no one's added it in the next month or so I'll likely take you up on that offer.  I appreciate the response and the package in general!

**Comment by ATL2001** [comment]
In spite of being completely new to typescript, I figured it out all on my own! I'll put in a PR later tonight for you to review when ya get a chance, hopefully you'll like it.   It took a lot less code than I had anticipated!  
![image](https://github.com/user-attachments/assets/50dd7ba3-caf2-48cc-a1c7-ea02a8f9ba08)
